applications/asset_tracker_v2/src/events/location_module_event.h

```diff
@@ -0,0 +1,167 @@
+ * @brief Location module event
+ * @defgroup location_module_event Location module event
+ * @{
+ */
+
+#include <nrf_modem_gnss.h>
+#include <modem/lte_lc.h>
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+#include <net/nrf_cloud_pgps.h>
+#endif
+#include <app_event_manager.h>
+#include <app_event_manager_profiler_tracer.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Location event types submitted by location module. */
+enum location_module_event_type {
+	/** A valid location has been obtained and the data is ready to be used.
+	 *
+	 *  The event has associated payload of the type ``struct location_module_data`` in
+	 *  the event struct member ``data.location``.
+	 *  All struct members within `data.location`` contain valid data.
+	 */
+	LOCATION_MODULE_EVT_DATA_READY,
+
+	/** Location could not be obtained and will not be ready for this
+	 *  sampling interval.
+	 *  The event has no associated payload.
+	 */
+	LOCATION_MODULE_EVT_DATA_NOT_READY,
+
+	/** Neighbor cell measurements have been gathered and the data is ready.
+	 *  The event has associated payload of type @ref location_module_neighbor_cells in
+	 *  the `data.neighbor_cells` member.
+	 */
+	LOCATION_MODULE_EVT_NEIGHBOR_CELLS_DATA_READY,
+
+	/** The location search timed out without acquiring location.
+	 *
+	 *  The event has associated payload of the type ``struct location_module_data`` in
+	 *  the event struct member ``data.location``.
+	 *  The ``data.location`` struct members ``search_time`` and ``satellites_tracked``
+	 *  contain valid data.
+	 */
+	LOCATION_MODULE_EVT_TIMEOUT,
+
+	/** An active location search has started. */
+	LOCATION_MODULE_EVT_ACTIVE,
+
+	/** A location search has stopped, either as a result of timeout, error
+	 *  or acquired location.
+	 */
+	LOCATION_MODULE_EVT_INACTIVE,
+
+	/** The module has been shut down gracefully.
+	 *  The event has associated payload of the type ``uint32`` in the struct member
+	 *  ``data.id``, which contains the module ID used when acknowledging a shutdown
+	 *  request from the util module.
+	 */
+	LOCATION_MODULE_EVT_SHUTDOWN_READY,
+
+	/** Location library has reported that it needs GPS assistance data.
+	 *  The event has associated payload of the type ``struct nrf_modem_gnss_agps_data_frame``
+	 *  in the event struct member ``data.agps_request``, which contains the types
+	 *  of A-GPS data that the modem needs.
+	 */
+	LOCATION_MODULE_EVT_AGPS_NEEDED,
+
+	/** Location library has reported that it needs GPS prediction data.
+	 *  The event has associated payload of the type ``struct gps_pgps_request``
+	 *  in the event struct member ``data.pgps_request``, which contains the types
+	 *  of P-GPS data that the modem needs.
+	 */
+	LOCATION_MODULE_EVT_PGPS_NEEDED,
+
+	/** An error has occurred, and data may have been lost.
+	 *  The event has associated payload of the type ``int`` in the struct member
+	 *  ``data.err``, which contains the original error code from that triggered
+	 *  the sending of this event.
```

Comment:
, which -> that
"... error code from that triggered the sending of this event" -> this is a bit unclear